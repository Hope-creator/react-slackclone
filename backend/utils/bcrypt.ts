import bcrypt from "bcrypt";
import { promisify } from "util";

const { genSalt, hash, compare } = bcrypt;

const promisifyGenSalt = promisify(genSalt);
const promisifyHash = promisify(hash);
const promisifyCompare = promisify(compare);

const hashText = (plainTextPw: string) => promisifyGenSalt().then((salt) => promisifyHash(plainTextPw, salt));

export { promisifyCompare, hashText };
