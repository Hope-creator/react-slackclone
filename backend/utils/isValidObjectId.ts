import mongoose from "mongoose";

export const idValidObjectId = (id: string) =>
  mongoose.Types.ObjectId.isValid(id);
