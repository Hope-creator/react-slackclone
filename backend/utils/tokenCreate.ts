import jwt from 'jsonwebtoken';

export const tokenCreate = (id: string) => {
     const token = jwt.sign(
        {
          userId: id,
        },
        (process.env.JWT_SECRET as string) || "random-secret-token",
        {
          expiresIn: 60 * 60 * 24 * 14,
        }
      );
      return token
}