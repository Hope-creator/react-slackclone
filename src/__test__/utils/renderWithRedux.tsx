import { render } from "@testing-library/react";
import { AnyAction, configureStore, EnhancedStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { rootReducer } from "../../store/rootReducer";

export interface IRenderParams {
  preloadedState: any;
  store: EnhancedStore;
}

const renderWithRedux = (
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  } = {} as IRenderParams
) => {
  const utils = {
    getState() {
      return store.getState();
    },
    dispatch(action: AnyAction) {
      return store.dispatch(action);
    },
  };
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    ...utils,
    ...renderOptions,
  };
};

export { renderWithRedux };
