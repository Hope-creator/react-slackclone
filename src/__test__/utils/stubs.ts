const stubUser = {
  away: false,
  company: { name: "My new company", _id: "T01TE7T5WEV" },
  conversations: [],
  createdAt: "2021-07-20T11:35:09.725Z",
  email: "Test@mail.com",
  is_admin: true,
  name: "Test",
  online: false,
  updatedAt: "2021-07-20T11:35:09.725Z",
  _id: "60f6b4ed0acbf2002bb5d29b",
  avatar: "some-src.com",
  display_name: "TestName",
};

export { stubUser };
