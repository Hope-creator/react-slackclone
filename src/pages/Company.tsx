import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useHistory } from "react-router";
import { selectUser } from "../store/modules/user/selectors";
import { fetchConversations } from "../store/modules/conversations/conversations";
import { selectConversations } from "../store/modules/conversations/selectors";

// # Components
import { RightSideBlock } from "../components/RightSideBlock";
import CompanyHeader from "../components/CompanyHeader";
import { CompanyMenuContent } from "../components/LeftSideBlock/CompanyMenuContent";
import { Navbar } from "../components/LeftSideBlock/Navbar";
import { CompanyMenuButton } from "../components/LeftSideBlock/CompanyMenuButton";

// #Mui
import Grid from "@material-ui/core/Grid";
import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";
import { Workspace } from "../components/Workspace/Workspace";
import {
  selectCurrentInfoItem,
  selectCurrentInfoItemType,
} from "../store/modules/currentInfo/selectors";
import { InfoItemTypeState } from "../store/modules/currentInfo/types";
import socket from "../services/socket/socket";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    companyWrapper: {
      height: "100vh",
    },
    companyContentWrapper: {
      height: "calc(100% - 38px)",
    },
    menuItemWrapper: {
      outline: "none",
      width: "300px",
      backgroundColor: "rgba(248,248,248,1)",
    },
    companyMenuButton: {
      backgroundColor: "#FBFAF7",
      height: 64,
      justifyContent: "start",
      fontWeight: 700,
      borderBottom: "1px solid rgb(233,227,230)",
      boxShadow: "none",
      color: theme.palette.primary.main,
      "&:hover": {
        backgroundColor: theme.palette.secondary.light,
      },
    },
    channelSidebar: {
      backgroundColor: "#FBFAF7",
    },
    workspace: {
      height: "100%",
      borderLeft: "1px solid rgb(233,227,230)",
      borderRight: "1px solid rgb(233,227,230)",
      maxHeight: "100%",
    },
    rightSideInfoWrapper: {
      overflowY: "auto",
      overflowX: "hidden",
      maxHeight: "100%",
      "&::-webkit-scrollbar": {
        width: "0.5em",
      },
      "&::-webkit-scrollbar-thumb": {
        backgroundColor: "rgba(29,28,29,.52)",
        borderRadius: "4px",
      },
    },
  })
);

export const Company = () => {
  const classes = useStyles();
  const user = useSelector(selectUser);
  const history = useHistory();
  const dispatch = useDispatch();
  const conversations = useSelector(selectConversations);
  const itemType = useSelector(selectCurrentInfoItemType);
  const itemInfo = useSelector(selectCurrentInfoItem);

  useEffect(() => {
    if (user) {
      dispatch(fetchConversations());
      socket.on("SERVER:CONVERSATION_CREATED", () => dispatch(fetchConversations()));
      socket.on("SERVER:NEW_MESSAGE", () => dispatch(fetchConversations()));
    }
    return () => {
      socket.removeListener("SERVER:DIALOG_CREATED", () =>
        dispatch(fetchConversations())
      );
      socket.removeListener("SERVER:NEW_MESSAGE", () =>
        dispatch(fetchConversations())
      );
    };
  }, [user, dispatch]);

  if (!user) {
    history.push("/get-started");
    return null;
  } else {
    const company = user.company;

    return (
      <div className={classes.companyWrapper}>
        <CompanyHeader user={user} />
        <Grid container wrap="nowrap" className={classes.companyContentWrapper}>
          <Grid item xs={3} className={classes.channelSidebar}>
            <CompanyMenuButton
              buttonClassName={classes.companyMenuButton}
              companyName={company.name}
            >
              <CompanyMenuContent company={company} />
            </CompanyMenuButton>
            <Navbar conversations={conversations} />
          </Grid>
          <Grid item xs className={classes.workspace}>
            <Workspace />
          </Grid>
          {itemType !== InfoItemTypeState.NONE && itemInfo !== undefined ? (
            <Grid className={classes.rightSideInfoWrapper} item xs={3}>
              <RightSideBlock item={itemInfo} type={itemType} />
            </Grid>
          ) : null}
        </Grid>
      </div>
    );
  }
};
