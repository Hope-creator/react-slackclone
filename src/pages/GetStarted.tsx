import { Link, makeStyles, Typography } from "@material-ui/core";
import React from "react";
import { GetStartedForm } from "../components/GetStartedForm";
import { LoginHeader } from "../components/LoginHeader";

const useStyles = makeStyles((theme) => ({
  wrapper: {
    textAlign: "center",
    height: "100vh",
    display: "flex",
    flexDirection: "column",
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  button: {
    fontWeight: 700,
    marginBottom: 20,
  },
  input: {
    margin: "0 0 20px",
  },
  signInForm: {
    maxWidth: "400px",
    marginTop: "40px",
    marginBottom: "20px",
    [theme.breakpoints.down("sm")]: {
      width: 200,
      marginTop: "20px",
    },
  },
  header: {
    [theme.breakpoints.down("sm")]: {
      width: 200,
      "& h3": {
        fontSize: "2rem",
      },
    },
  },
  inputFocused: {
    transition: theme.transitions.create(["border", "box-shadow"], {
      duration: "80ms",
    }),
    boxShadow: "0px 0px 0px 4px rgba(29,155,209, 0.3)",
  },
  radioCheckWrapper: {
    textAlign: "start",
  },
  formLegend: {
    fontSize: "0.9rem",
    color: "#1d1c1d",
    fontWeight: 700,
  },
  formTsAndCs: {
    textAlign: "start",
    margin: "20px 0 70px 0",
    color: "#696969",
    "&>a": {
      color: "#696969",
    },
  },
}));

export const GetStarted: React.FC = (): React.ReactElement => {
  const classes = useStyles();

  return (
    <div className={classes.wrapper}>
      <div className={classes.header}>
        <LoginHeader newToSlackShow={false} title={"First, enter your email"} />
      </div>
      <div className={classes.signInForm}>
        <div className={classes.signInForm}>
          <GetStartedForm />
        </div>
        <Typography variant="body2" className={classes.formTsAndCs}>
          By continuing, youâ€™re agreeing to our{" "}
          <Link>Customer Terms of Service</Link>, <Link>Privacy Policy</Link>,
          and <Link>Cookie Policy</Link>.
        </Typography>
      </div>
    </div>
  );
};
