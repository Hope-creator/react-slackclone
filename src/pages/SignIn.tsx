import {
  Button,
  FormControlLabel,
  FormLabel,
  Link,
  makeStyles,
  Radio,
  RadioGroup,
  TextField,
  Typography,
} from "@material-ui/core";
import React from "react";
import { LoginFooter } from "../components/LoginFooter";
import { LoginHeader } from "../components/LoginHeader";

const useStyles = makeStyles((theme) => ({
  wrapper: {
    textAlign: "center",
    height: "100vh",
    display: "flex",
    flexDirection: "column",
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  button: {
    fontWeight: 700,
    marginBottom: 20,
  },
  input: {
    margin: "0 0 20px",
  },
  signInForm: {
    maxWidth: "400px",
    marginTop: "40px",
    marginBottom: "20px",
  },
  inputFocused: {
    transition: theme.transitions.create(["border", "box-shadow"], {
      duration: "80ms",
    }),
    boxShadow: "0px 0px 0px 4px rgba(29,155,209, 0.3)",
  },
  radioCheckWrapper: {
    textAlign: "start",
  },
  formLegend: {
    fontSize: "0.9rem",
    color: "#1d1c1d",
    fontWeight: 700,
  },
  formTsAndCs: {
    textAlign: "start",
    margin: "20px 0 70px 0",
    color: "#696969",
    "&>a": {
      color: "#696969"
    }
  },
}));

export const SignIn = () => {
  const classes = useStyles();

  return (
    <div className={classes.wrapper}>
      <LoginHeader newToSlackShow={false} title={"First, enter your email"} />
      <div className={classes.signInForm}>
        <TextField
          className={classes.input}
          InputProps={{
            classes: {
              focused: classes.inputFocused,
            },
          }}
          label="name@work-email.com"
          id="outlined-basic"
          color="secondary"
          size="small"
          variant="outlined"
          fullWidth
        />
        <Button
          className={classes.button}
          variant="contained"
          color="primary"
          fullWidth
        >
          Continue
        </Button>
        <FormLabel className={classes.formLegend} component="legend">
          Can we send you email with Slack tips, news, and offers?
        </FormLabel>
        <RadioGroup
          aria-label="Can we send you email with Slack tips, news, and offers?"
          name="emailoffers"
        >
          <FormControlLabel
            value="Sure!"
            control={<Radio size="small" />}
            label="Sure!"
          />
          <FormControlLabel
            value="No thanks"
            control={<Radio size="small" />}
            label="No thanks"
          />
        </RadioGroup>
        <Typography variant="body2" className={classes.formTsAndCs}>
          By continuing, youâ€™re agreeing to our{" "}
          <Link>Customer Terms of Service</Link>, <Link>Privacy Policy</Link>,
          and <Link>Cookie Policy</Link>.
        </Typography>
      </div>

      <LoginFooter />
    </div>
  );
};
