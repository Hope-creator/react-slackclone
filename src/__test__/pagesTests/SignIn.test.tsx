import { SignIn } from "../../pages/SignIn";
import { MemoryRouter } from "react-router-dom";
import { renderWithRedux } from "../utils/renderWithRedux";

describe("GetStarted page tests", () => {
  it("should current render component", () => {
    const { asFragment } = renderWithRedux(
      <MemoryRouter initialEntries={["/signin"]}>
        <SignIn />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
