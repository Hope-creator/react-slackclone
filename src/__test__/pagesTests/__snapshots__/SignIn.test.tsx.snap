// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted page tests should current render component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-wrapper-1"
  >
    <div
      class="makeStyles-header-13"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-end"
        >
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            New to Slack?
          </span>
          <div
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-caption MuiTypography-colorSecondary"
          >
            <a
              href="/get-started"
            >
              Create an account
            </a>
          </div>
        </div>
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          Sign in to Slack
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          We suggest using the 
          <strong>
            email address you use at work.
          </strong>
        </p>
      </div>
    </div>
    <div
      class="makeStyles-signInForm-6"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-14 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
            for="email-outlined-basic"
            id="email-outlined-basic-label"
          >
            name@work-email.com
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="email-outlined-basic"
              name="email"
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19"
              >
                <span>
                  name@work-email.com
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-14 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Your password
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="outlined-basic"
              name="password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-19"
              >
                <span>
                  Your password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-16 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Sign in with Email
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
      <div
        class="makeStyles-emailPrompt-7"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-emailPromptIcon-8"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 11H1v2h6v-2zm2.17-3.24L7.05 5.64 5.64 7.05l2.12 2.12 1.41-1.41zM13 1h-2v6h2V1zm5.36 6.05l-1.41-1.41-2.12 2.12 1.41 1.41 2.12-2.12zM17 11v2h6v-2h-6zm-5-2c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3zm2.83 7.24l2.12 2.12 1.41-1.41-2.12-2.12-1.41 1.41zm-9.19.71l1.41 1.41 2.12-2.12-1.41-1.41-2.12 2.12zM11 23h2v-6h-2v6z"
          />
        </svg>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          Weâ€™ll email you a magic code for a password-free sign in. Or you can 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
          >
            sign in manually instead.
          </a>
        </h6>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
